js/src/forum/states/DiscussionListState.js

```diff
@@ -77,17 +77,21 @@ export default class DiscussionListState {
   }
 
   /**
-   * Clear and reload the discussion list.
+   * Clear and reload the discussion list. Passing the option `deferClear: true`
+   * will clear discussions only after new data has been received.
+   * This can be used to refresh discussions without loading animations.
    */
-  refresh({ clear = true } = {}) {
+  refresh({ deferClear = false } = {}) {
     this.loading = true;
 
-    if (clear) {
+    if (!deferClear) {
       this.clear();
     }
 
     return this.loadResults().then(
       (results) => {
+        // We don't use the clear method, since it would lead to an additional redraw.
```

Comment:
Just one last suggestion, I'd make this comment say something like "Same effect as this.clear(), but without an additional redraw"

I find the current phrasing a bit confusing.

Otherwise, nothing else to add.